src/components/BSelect/BSelect.vue

```diff
@@ -0,0 +1,283 @@
+<template>
+  <div>
+    <div>
+      <label :id="'label-' + id">{{ selectLabel }} <span v-if=required aria-label='Required'>*</span></label>
+    </div>
+    <button
+      :class="dropdownClass"
+      :aria-expanded="opened"
+      :aria-owns="id"
+      :aria-required="required"
+      :aria-describedby="'error' + id"
+      :aria-labelledby="'label-' + id + ' dropdown-' + id"
+      :id="'dropdown-' + id"
+      :ref="'dropdown-' + id"
+      @keyup.space.prevent.stop="toggleList"
+      @keyup.enter.prevent.stop="toggleList"
+      @click="toggleList"
+      tabindex="0"
+      v-model="selectLabel"
+      aria-haspopup="listbox"
+      >{{ selectedOption ? selectedOption.text : placeholder }}
+      <span class="dropdown-arrow"><i class="fa fa-caret-down" aria-hidden="true"></i></span>
+    </button>
+    <label hidden
+    <ul
+      :class="ulClass"
+      :id="id"
+      role="listbox"
+    >
+      <li
+        v-if="opened"
+        v-for="option in selectOptions"
+        :aria-selected="selectedOption && selectedOption.id === option.id"
+        :id="'option-' + option.id"
+        :key="'selectOption' + option.id"
+        :value="option.id"
+        :ref="'option-' + option.id"
+        @click="selectOption(option)"
+        @keyup.up.prevent="upHandler($event)"
+        @keyup.down.prevent="downHandler($event)"
+        @keyup.enter.prevent="selectOption(option)"
+        @keyup.space.prevent.stop="selectOption(option)"
```

Comment:
Ok, last thing I swear `@keyup.esc="toggleList"` should be added to this so that users can close the menu and is there an `@blur` event that can also be bound to "toggleList" so that when focus leaves the menu it closes?